src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -585,4 +585,30 @@ public void ListenNamedPipe(string pipeName, Action<ListenOptions> configure)
         configure(listenOptions);
         CodeBackedListenOptions.Add(listenOptions);
     }
+
+    /// <summary>
+    /// Return true if there is a default or development server certificate.
+    /// Should not be called before the configuration is loaded, if there is one.
+    /// </summary>
+    /// <returns>True if there is a default or development server certificate, false otherwise.</returns>
+    /// <exception cref="InvalidOperationException">If there is a configuration and it has not been loaded.</exception>
+    public bool HasDefaultOrDevelopmentCertificate
```

Comment:
Not super enthusiastic about a property getter having side effects.

Reply:
Yeah, make it a method.

Reply:
What side-effect are we worried about?  Enabling https config?

Reply:
I'm not familiar with the `bool CheckFoo()` naming convention - personally, I'd expect that method to throw if the condition weren't satisfied.  I take it we use that elsewhere?

Reply:
Renamed and made it a method, but I'd still like to understand more about side-effects.

Reply:
The point of the method is to avoid throwing for known, common conditions. Exceptions should only be used for unexpected conditions.

Reply:
Incidentally, now that it doesn't actually return the cert, I would be fine with this throwing if https configuration isn't enabled.

Reply:
> The point of the method is to avoid throwing for known, common conditions. Exceptions should only be used for unexpected conditions.

Sorry, I'm not sure which part of the thread you're responding to.  If it was about my expectation that `CheckFoo` would throw, that was an intuition about the naming convention and not a comment on how this member (whatever name and shape it ends up taking) should behave.

Reply:
This is a friendly guard method, letting you check the state in a controlled way to avoid an exception later. It should not throw an exception.

Reply:
@halter73 Asked me to have it throw in that case. Would you like me to work up a version that loads eagerly and then compensates if the configuration changes?

Reply:
If you're getting conflicting advice we should hold off until we can discuss it in a meeting, API review or otherwise.